sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -480,10 +480,120 @@ private[sql] object DataSourceStrategy extends Strategy with Logging {
       case expressions.Contains(a: Attribute, Literal(v: UTF8String, StringType)) =>
         Some(sources.StringContains(a.name, v.toString))
 
+      case expressions.BinaryComparison(BinaryArithmetic(left, right), Literal(v, t)) =>
+        translateArithemiticOPFilter (predicate)
+      case expressions.BinaryComparison(Literal(v, t), BinaryArithmetic(left, right)) =>
+        translateArithemiticOPFilter (predicate)
+        
       case _ => None
     }
   }
 
+  private def translateArithemiticOPFilter(predicate: Expression): Option[Filter] = {
+    predicate match {
+      case expressions.EqualTo(Add(left, right), Literal(v, t)) =>
+        Some(sources.ArithmeticOPEqualTo(Add(left, right), convertToScala(v, t)))
```

Comment:
Hm.. I see. but for me it looks this might have to hide `expression._`. In this way, the `expression._` can be accessed in datasource level. I believe the reason way `source._` is implemented is, to hide `expression._` which has been changed rapidly version by version.